solr/core/src/java/org/apache/solr/search/join/BlockJoinChildQParser.java

```diff
@@ -44,10 +45,8 @@ protected String getParentFilterLocalParamName() {
   @Override
   protected Query noClausesQuery() throws SyntaxError {
     final Query parents = parseParentFilter();
-    final BooleanQuery notParents = new BooleanQuery.Builder()
-        .add(new MatchAllDocsQuery(), Occur.MUST)
-        .add(parents, Occur.MUST_NOT)
-      .build();
-    return new BitSetProducerQuery(getBitSetProducer(notParents));
+    final BooleanQuery.Builder notParentsBuilder = new BooleanQuery.Builder().add(new MatchAllDocsQuery(), Occur.MUST);
+    final BitSetProducerQuery notParentsQuery=(parents!=null)?  new BitSetProducerQuery(getBitSetProducer(notParentsBuilder.add(parents, Occur.MUST_NOT).build())): new BitSetProducerQuery(getBitSetProducer(notParentsBuilder.build()));
+    return notParentsQuery;
```

Comment:
This is very hard to read, can you add some whitespace here or possibly switch to using a proper if statement instead of ternary conditional.